src/signal.rs

```diff
@@ -667,6 +696,135 @@ mod widget {
 
 }
 
+pub trait EntrySignals {
+    fn connect_activate<F: Fn(Entry) + 'static>(&self, f: F) -> u64;
+    fn connect_backspace<F: Fn(Entry) + 'static>(&self, f: F) -> u64;
+    fn connect_copy_clipboard<F: Fn(Entry) + 'static>(&self, f: F) -> u64;
+    fn connect_cut_clipboard<F: Fn(Entry) + 'static>(&self, f: F) -> u64;
+    fn connect_paste_clipboard<F: Fn(Entry) + 'static>(&self, f: F) -> u64;
+    fn connect_toggle_overwrite<F: Fn(Entry) + 'static>(&self, f: F) -> u64;
+    fn connect_delete_from_cursor<F: Fn(Entry, DeleteType, i32) + 'static>(&self, f: F) -> u64;
+    fn connect_move_cursor<F: Fn(Entry, MovementStep, i32, bool) + 'static>(&self, f: F) -> u64;
+    fn connect_insert_at_cursor<F: Fn(Entry, &str) + 'static>(&self, f: F) -> u64;
```

Comment:
Turns out `Fn(&str)` is sugar for `for<'r> Fn(&'r str)` which means that the closure is generic over the lifetime of its argument, and (as I see it) since it has to accept arbitrarily short `'r`s it's safe to pass in our short-lived reference.